graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -673,13 +867,82 @@ def _rename_or_suppress_or_ignore_name_and_add_to_record(
         if desired_type_name in builtin_scalar_type_names:
             self.type_renamed_to_builtin_scalar_conflicts[type_name] = desired_type_name
 
+        # Any potential type suppressions will have taken place by this point, so this current node
+        # will appear in the renamed schema, so it's safe to apply field renamings to this type.
+        fields_renamed_node = node  # If no field renaming happens, fields_renamed_node will just be
+        # the current node, unchanged.
+        if isinstance(fields_renamed_node, ObjectTypeDefinitionNode):
+            # mypy is unable to detect that fields_renamed_node is an ObjectTypeDefinitionNode if
+            # the code enters this block, so disabling it for this line.
+            # https://github.com/python/mypy/issues/2885#issuecomment-287928126
+            fields_renamed_node = self._rename_fields(fields_renamed_node)  # type: ignore
         self.reverse_name_map[desired_type_name] = type_name
         if desired_type_name == type_name:
-            return IDLE
+            return fields_renamed_node
         else:  # Make copy of node with the changed name, return the copy
-            node_with_new_name = get_copy_of_node_with_new_name(node, desired_type_name)
+            node_with_new_name = get_copy_of_node_with_new_name(
+                fields_renamed_node, desired_type_name
+            )
             return node_with_new_name
 
+    def _rename_fields(self, node: ObjectTypeDefinitionNode) -> ObjectTypeDefinitionNode:
+        """Rename node's fields, if applicable and return new node with updated fields."""
+        type_name = node.name.value
+        current_type_field_renamings = self.field_renamings.get(type_name, None)
+        if current_type_field_renamings is None:
```

Comment:
Since current_type_field_renamings need not be a dict, I allow for it to be None if we don't want to do any field renamings for the current type. However, that means it'd technically be valid for it to be an empty dict and I'm not sure how to best check for this, especially since it's not the falsiness of current_type_field_renamings that we're looking for but rather whether it contains any field renamings.